### PR TITLE
Converted README.txt to markdown, add TOC

### PR DESCRIPTION
**Pull request by Minimalistic**
Converted README.txt to README.md, add a Table of Contents with relative linking and reformat the readme to take advantage of markdown formatting styles.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -327,8 +359,8 @@ Compiling on Windows:
 	- Select your compiler
 	- It will warn about missing stuff, ignore that at this point. (later don't)
 	- Make sure the configuration is as follows
-	  (note that the versions may differ for you):
-	-----------------
+	### (note that the versions may differ for you):
```
Comment (SmallJoker) [inline README.md]: This is not supposed to be a title. The dashes should be replaced with a code block.

File: README.md [hunk 2]
```diff
@@ -342,12 +374,12 @@ Compiling on Windows:
 	GETTEXT_BIN_DIR          DIR/gettext/bin
 	GETTEXT_INCLUDE_DIR      DIR/gettext/include
 	GETTEXT_LIBRARIES        DIR/gettext/lib/intl.lib
-	GETTEXT_MSGFMT           DIR/gettext/bin/msgfmt
-	-----------------
+	## GETTEXT_MSGFMT           DIR/gettext/bin/msgfmt
```
Comment (SmallJoker) [inline README.md]: Code block ends here

File: README.md [hunk 3]
```diff
@@ -398,59 +431,59 @@ exit /b 0
...
-ShadowNinja:
+##### ShadowNinja:
   textures/base/pack/smoke_puff.png
 
-Paramat:
+##### Paramat:
   textures/base/pack/menu_header.png
 
-erlehmann:
+#####erlehmann:
```
Comment (SmallJoker) [inline README.md]: Missing space

File: README.md [hunk 4]
```diff
@@ -398,59 +431,59 @@ exit /b 0
...
-
-Irrlicht
----------------
+## Irrlicht
 
-This program uses the Irrlicht Engine. http://irrlicht.sourceforge.net/
+This program uses the [Irrlicht Engine.](http://irrlicht.sourceforge.net/)
 
- The Irrlicht Engine License
+#####The Irrlicht Engine License
```
Comment (SmallJoker) [inline README.md]: Another missing space

File: README.md [hunk 5]
```diff
@@ -372,8 +404,9 @@ Compiling on Windows:
 		- You should now have a working game with the executable in
 			DIR/minetest/bin/minetest.exe
 
-Windows releases of minetest are built using a bat script like this:
---------------------------------------------------------------------
+## Windows Releases Note:
```
Comment (kaeza) [inline README.md]: Release Notes*

File: README.md [hunk 6]
```diff
@@ -398,59 +431,59 @@ exit /b 0
...
 
-Authors of media files
------------------------
+## Authors of Media Files
+
 Everything not listed in here:
 Copyright (C) 2010-2012 celeron55, Perttu Ahola <celeron55@gmail.com>
 
-ShadowNinja:
+##### ShadowNinja:
```
Comment (kaeza) [inline README.md]: Use a list:

```
* somedev:
    * `foo/bar.png`
    * `foo/baz.png`
* someotherdev:
    * `notbad.jpg`
```

File: README.md [hunk 7]
```diff
@@ -398,59 +431,59 @@ exit /b 0
...
-You should have received a copy of the GNU Lesser General Public License along
-with this program; if not, write to the Free Software Foundation, Inc.,
-51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
-
-Irrlicht
----------------
+## Irrlicht
 
-This program uses the Irrlicht Engine. http://irrlicht.sourceforge.net/
+This program uses the [Irrlicht Engine.](http://irrlicht.sourceforge.net/)
```
Comment (kaeza) [inline README.md]: Period should be outside of the link text:

    This program uses the [Irrlicht Engine](http://irrlicht.sourceforge.net/).

File: README.md [hunk 8]
```diff
@@ -1,58 +1,91 @@
...
+    * [Irrlicht](#irrlicht)
+    * [JThread](#jthread)
+    * [Lua](#lua)
+    * [Fonts](#fonts)
+
+## In Case You Downloaded The Source Code:
+
 If you downloaded the Minetest Engine source code in which this file is
 contained, you probably want to download the minetest_game project too:
   https://github.com/minetest/minetest_game/
```
Comment (kaeza) [inline README.md]: Make this a link?

File: README.md [hunk 9]
```diff
@@ -1,58 +1,91 @@
...
 See the README.txt in it.
 
-Further documentation
-----------------------
+## Further Documentation
+
 - Website: http://minetest.net/
```
Comment (kaeza) [inline README.md]: Make these links?

**Comment by Minimalistic** [comment]
Thanks for the feedback!  I'll get to work on the requested adjustments.

File: README.md [hunk 10]
```diff
@@ -69,103 +100,125 @@ Default controls
...
+`sudo dnf install git`
 
-For Fedora users:
-$ sudo dnf install git
+- Download source (this is the URL to the latest of source repository, which might not work at all times) using git:
 
-Download source (this is the URL to the latest of source repository, which might not work at all times) using git:
-$ git clone --depth 1 https://github.com/minetest/minetest.git
-$ cd minetest
+`git clone --depth 1 https://github.com/minetest/minetest.git`
```
Comment (kaeza) [inline README.md]: These should be code blocks.

File: README.md [hunk 11]
```diff
@@ -69,103 +100,125 @@ Default controls
...
 Download source, without using git:
-$ wget https://github.com/minetest/minetest/archive/master.tar.gz
-$ tar xf master.tar.gz
-$ cd minetest-master
+
+`wget https://github.com/minetest/minetest/archive/master.tar.gz`
+
+`tar xf master.tar.gz`
+
+`cd minetest-master`
```
Comment (kaeza) [inline README.md]: See above.

File: README.md [hunk 12]
```diff
@@ -69,103 +100,125 @@ Default controls
...
+
+`cd games/`
+
+`wget https://github.com/minetest/minetest_game/archive/master.tar.gz`
+
+`tar xf master.tar.gz`
+
+`mv minetest_game-master minetest_game`
+
+`cd ..`
```
Comment (kaeza) [inline README.md]: Code block.

File: README.md [hunk 13]
```diff
@@ -69,103 +100,125 @@ Default controls
...
 
 Build a version that runs directly from the source directory:
-$ cmake . -DRUN_IN_PLACE=TRUE
-$ make -j <number of processors>
+
+`cmake . -DRUN_IN_PLACE=TRUE`
+
+`make -j <number of processors>`
```
Comment (kaeza) [inline README.md]: Code block.

File: README.md [hunk 14]
```diff
@@ -398,37 +450,13 @@ exit /b 0
...
------------------------
-Everything not listed in here:
-Copyright (C) 2010-2012 celeron55, Perttu Ahola <celeron55@gmail.com>
-
-ShadowNinja:
-  textures/base/pack/smoke_puff.png
-
-Paramat:
-  textures/base/pack/menu_header.png
+# Licensing
```
Comment (kaeza) [inline README.md]: This should be level 2 heading, and all headers logically "below" this one should be one more level.

**Comment by Minimalistic** [comment]
I apologize, I'm still learning the proper Github workflow - I implemented the simple changes of replacing 'code block' with the more appropriate ``` code block ```.  However I didn't create a new pull request, did I do this right or should it ideally have been a separate new pull request?

**Comment by est31** [comment]
Quite similar to #4968. No idea why it stalled

File: README.md [hunk 15]
```diff
@@ -243,92 +308,88 @@ ZLIB_DLL                        - Only on Windows; path to zlib1.dll
...
+- And, of course, [Minetest](http://minetest.net/download)
+
+### Steps:
+
+- Select a directory called DIR hereafter in which you will operate.
+- Make sure you have CMake and a compiler installed.
+- Download all the other stuff to DIR and extract them into there.
+  ("extract here", not "extract to packagename/")
+  NOTE: zlib125dll.zip needs to be extracted into zlib125dll
+  NOTE: You need to extract sqlite3.h & sqlite3ext.h from sqlite3 source
```
Comment (SmallJoker) [inline README.md]: This is not formatted proplerly, please use
```
    - NOTE: 
    - NOTE: 
```

File: README.md [hunk 16]
```diff
@@ -69,103 +102,135 @@ Default controls
...
+    - share = `Contents/Resources`
+    - user  = `Contents/User` OR `~/Library/Application Support/minetest`
+
+## World Directory
+
 - Worlds can be found as separate folders in:
-    $user/worlds/
+    - `user/worlds/`
+
+## Configuration File:
```
Comment (SmallJoker) [inline README.md]: Colon looks weird here, please remove them from all titles

File: README.md [hunk 17]
```diff
@@ -497,8 +560,7 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 IN THE SOFTWARE.
 
-Lua
----------------
+### Lua
```
Comment (SmallJoker) [inline README.md]: `##`, as the other libraries

**Comment by kaeza** [comment]
>I apologize, I'm still learning the proper Github workflow - [...] However I didn't create a new pull request, did I do this right or should it ideally have been a separate new pull request?

It should be separate commits under the same PR. Once it's considered "done" it can be rebased by the devs as needed.

**Comment by rubenwardy** [comment]
#6244

**Pull request closed without merging**